lib/super_digit.rb

```diff
@@ -1,15 +1,33 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n), with n being the number of digits in the initial integer.
+# Space Complexity - O(n), because of the recursive stack; size of the stack depends on the number of digits.
 def super_digit(n)
-   
+  sum = 0
+
+  if n < 10
+    return sum + n
+  else
+    until n < 10
+      sum += (n % 10)
+      n /= 10
+    end
+
+    sum += n
+  end
+
+  return super_digit(n = sum)
```

Comment:
You don't need to do the assignment here. 

```suggestion
  return super_digit(sum)
```